### PR TITLE
Fix flexi nights calculation

### PR DESCRIPTION
**Pull request by zediah**
The calculations that were done to generate the options for a package had two errors:
- The `packageId` was set incorrectly to the `offerPackage.id` - it should've been the `id_salesforce_external`
- Tours were returning `NaN` prices, this was due to `extraNights` being undefined and causing the calculation to fail.

Refactored the functions to make them a bit smaller.

All the actual work is done in the flexiNights files.

Other changes:
- Updated babel to latest, also supports latest syntax
- Updated eslint to latest, turns out this made quite a formatting changes...

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,22 +18,23 @@
     "preversion": "yarn run lint",
     "ci": "yarn test && yarn run lint && yarn build"
   },
-  "dependencies": {},
+  "dependencies": {
+    "@babel/preset-env": "^7.14.7"
```
Comment (keithbro) [inline package.json:22]: Is this not a devDependency?

**Review by keithbro — APPROVED** [review]
Looks good and thanks for modernizing it.

Just one question about having @babel/preset-env as a prod dependency rather than a dev dependency

**Comment by keithbro** [comment]
Oh I think you'll want to bump the version in package.json here so you can then use Github releases to publish it.

File: package.json (hunk 1 above)
Comment (zediah) [reply package.json:22]: oops, it is, I just did a `yarn add` and it put it there

**Pull request merged**